src/core/gui.py

```diff
@@ -219,6 +232,12 @@ def run(self):
 
 # Test the GUI with random numbers generation
```

Comment:
shouldn't this go into tests?

Reply:
This part is meant to visually assess the plot to assist you in the development, being able to launch the plot without calling APIs and launching directly to the plot window. Maybe after major development steps are done for the plot we can think of move/remove it.